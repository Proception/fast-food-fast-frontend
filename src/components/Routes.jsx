import React from "react";
import Home from "./Home.jsx";
import Login from "./Login.jsx";
import Logout from "./Logout.jsx";
import NotFound from "./NotFound.jsx"
import Restaurant from "./Restaurant.jsx";
import Signup from './Signup.jsx';
import { Route, Switch } from 'react-router-dom';
// import PrivateRoute from './PrivateRoute.jsx';
import AddFoodItems from './AddFoodItems.jsx';
import Checkout from './Checkout.jsx';
import Orders from './Orders.jsx';
import OrderHistory from './OrderHistory.jsx';

const Routes = () => (
    <Switch>
        <Route exact path="/" component={Home}></Route>
        <Route path="/login" component={Login}></Route>
        <Route path="/logout" component={Logout}></Route>
        <Route path="/signup" component={Signup}></Route>
        <Route path="/menu" component={Restaurant}></Route>
        <Route path="/addfooditems" component={AddFoodItems}></Route>
        <Route path="/checkout" component={Checkout}></Route>
        <Route path="/orders" component={Orders}></Route>
        <Route path="/orderhistory" component={OrderHistory}></Route>

        {/* <PrivateRoute  exact path="/menu" component={Restaurant}></PrivateRoute> */}
        <Route component={NotFound}></Route>
    </Switch>
);

export default Routes;
