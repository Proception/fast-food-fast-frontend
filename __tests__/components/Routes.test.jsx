import React from 'react';
import { shallow } from 'enzyme';
import Routes from '../../src/components/Routes.jsx';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Routes/>);
});
describe('Routes Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
