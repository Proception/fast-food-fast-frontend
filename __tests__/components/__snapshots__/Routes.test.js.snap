// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component should match snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Route
    component={[Function]}
    path="/logout"
  />
  <Route
    component={[Function]}
    path="/signup"
  />
  <Route
    component={[Function]}
    path="/menu"
  />
  <Route
    component={[Function]}
    path="/addfooditems"
  />
  <Route
    component={[Function]}
    path="/checkout"
  />
  <Route
    component={[Function]}
    path="/orders"
  />
  <Route
    component={[Function]}
    path="/orderhistory"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
